### PR TITLE
feat(Manager) : environment filter in spa detailed page

### PR DESCRIPTION
**Pull request by pjain24**
## Closes / Fixes / Resolves

Closes #issue_id

## Explain the feature/fix

I have implemented an "Environment Filter" on the SPA Detailed Page, which allows stakeholders to filter and view all SPAs within a specific environment. This feature provides a convenient way for stakeholders to search and access relevant information, enhancing the user experience.

## Does this PR introduce a breaking change

No

## Screenshot(s)

<details>
<summary>View Screenshots</summary>

Screenshot from 2023-10-25 15-04-53

Screenshot from 2023-10-25 15-05-07
Screenshot from 2023-10-25 15-05-15

</details>

### CODE REVIEW

File: packages/manager/src/views/WebPropertyDetailPage/components/SSR/ContainerizedDeployment.tsx [hunk 1]
```diff
@@ -289,225 +299,276 @@ export const ContainerizedDeployment = () => {
...
-                )} ${val?.path.length > URL_LENGTH_LIMIT ? '...' : ''}`}</Td>
-                <Td style={{ maxWidth: '15ch', wordWrap: 'break-word' }}>
-                  {`${val?.healthCheckPath.slice(0, URL_LENGTH_LIMIT)} ${
-                    val?.healthCheckPath.length > URL_LENGTH_LIMIT ? '...' : ''
-                  }`}
-                </Td>
+        <SplitItem isFilled>
+          <Select
+            width={300}
+            // style={{ width: '300px!important' }}
```
Comment (soumyadip007) [inline packages/manager/src/views/WebPropertyDetailPage/components/SSR/ContainerizedDeployment.tsx:509]: Please remove the comment

File: packages/manager/src/views/WebPropertyDetailPage/components/SSR/StaticDeployment.tsx [hunk 2]
```diff
@@ -85,213 +96,294 @@ export const StaticDeployment = () => {
...
-              <Th textCenter>Router URL</Th>
-              <Th textCenter style={{ justifyContent: 'space-evenly', display: 'grid' }}>
-                Actions
-              </Th>
-            </Tr>
-          </Thead>
+    <>
+      <Split hasGutter className="pf-u-mb-md">
+        <SplitItem isFilled>
+          <Select
```
Comment (nikhitamore) [inline packages/manager/src/views/WebPropertyDetailPage/components/SSR/StaticDeployment.tsx:124]: @pjain24 Can you try writing a separate component EnvFilter for this , and call it in containerized and static, as code is being repeated in both the files.
Create a component that accepts data and returns a dropdown

File: packages/manager/src/views/WebPropertyDetailPage/components/SSR/StaticDeployment.tsx [hunk 3]
```diff
@@ -85,213 +96,294 @@ export const StaticDeployment = () => {
...
+                    <Td textCenter>
+                      {' '}
+                      {`${val?.name.slice(0, URL_LENGTH_LIMIT)} ${
+                        val?.name.length > URL_LENGTH_LIMIT ? '...' : ''
+                      }`}
+                    </Td>
+                    <Td textCenter>
                       <Label
-                        key={`autoSync-${val.name}`}
+                        key={val.env}
```
Comment (nikhitamore) [inline packages/manager/src/views/WebPropertyDetailPage/components/SSR/StaticDeployment.tsx:230]: May I ask why was this key changed, {`autoSync-${val.name}`} to env

**Review by nikhitamore — CHANGES_REQUESTED** [review]

File: packages/manager/src/views/Settings/components/Symlink/Symlink.tsx [hunk 4]
```diff
@@ -39,8 +39,9 @@ type EnvItem = {
   isActive: boolean;
   createdAt: string;
   updatedAt: string;
-  symlink: TSymlink;
+  symlink: [];
```
Comment (nikhitamore) [inline packages/manager/src/views/Settings/components/Symlink/Symlink.tsx:42]: Why was TSymlink removed

**Comment by soumyadip007** [comment]
Hi @pjain24 , I can see a lot of API changes and Manager Overridden code in the PR. Please rebase it.

File: packages/api/src/repository/mongo/model/environment.model.ts [hunk 5]
```diff
@@ -39,7 +39,7 @@ export class Environment {
   createdBy: string;
 
   @Prop({})
-  symlink: Symlink;
+  symlink: Symlink[];
 
   @Prop({ default: 'NA' })
```
Comment (soumyadip007) [inline packages/api/src/repository/mongo/model/environment.model.ts:44]: This is API change, and already released, please rebase it.

File: packages/api/package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "spaship-api",
-  "version": "4.0.0",
```
Comment (soumyadip007) [inline packages/api/package.json:3]: This is API change, and already released, please rebase it.

File: packages/api/src/server/analytics/activity-stream.entity.ts [hunk 7]
```diff
@@ -46,6 +46,7 @@ export enum Action {
   ENV_UPDATED = 'ENV_UPDATED',
   ENV_DELETED = 'ENV_DELETED',
   ENV_SYNCED = 'ENV_SYNCED',
+  SYMLINK_CREATED = 'SYMLINK_CREATED',
   WEBHOOK_CREATED = 'WEBHOOK_CREATED',
   WEBHOOK_DELETED = 'WEBHOOK_DELETED',
   WEBHOOK_UPDATED = 'WEBHOOK_UPDATED',
```
Comment (soumyadip007) [inline packages/api/src/server/analytics/activity-stream.entity.ts:52]: This is an API change, and already released, please rebase it.

File: packages/api/src/server/application/service/application.service.ts [hunk 8]
```diff
@@ -319,6 +319,21 @@ export class ApplicationService {
...
+          { propertyIdentifier, env, identifier, isContainerized: true, isGit: false },
+          applicationDetails
+        );
+        return this.applicationFactory.createApplicationResponse(applicationDetails, applicationExists);
+      }
+      if (JSON.stringify(applicationRequest.config) !== JSON.stringify(applicationDetails.config)) {
+        const applicationConfigDTO = this.applicationFactory.configRequestBuilder(propertyIdentifier, identifier, env, applicationRequest);
+        await this.saveConfig(applicationConfigDTO);
+      }
       applicationDetails.nextRef = this.applicationFactory.getNextRef(applicationRequest.ref) || 'NA';
```
Comment (soumyadip007) [inline packages/api/src/server/application/service/application.service.ts:337]: This is API change, and already been released, please rebase it.

**Comment by nikhitamore** [comment]
Closing as created 2520 with proper rebasing and code

**Pull request closed without merging**